pkg/operator/ceph/disruption/nodedrain/reconcile.go

```diff
@@ -191,3 +249,35 @@ func newDoNothingContainers(rookImage string) []corev1.Container {
 		Args:    []string{"--", "sleep", "infinity"},
 	}}
 }
+
+func getDeploymentForPod(client client.Client, podKey types.NamespacedName) (*appsv1.Deployment, error) {
+	// get pod
+	operatorPod := &corev1.Pod{}
+	err := client.Get(context.TODO(), podKey, operatorPod)
+	if err != nil {
+		return nil, fmt.Errorf("could not get pod %+v: %+v", podKey, err)
+	} else {
+		for _, rsRef := range operatorPod.ObjectMeta.OwnerReferences {
+			if *rsRef.Controller {
```

Comment:
why must the `Controller` property be set when querying the owner references of the parent replicaset or deployment?

Reply:
The pod is  controlled by the replicaset, and the replicaset is controlled by the deployment.